### DISCUSSION TITLE
Need a way to install unison 2.48.4

### DISCUSSION THREAD

**Discussion by monk-wellington** [Homebrew/discussions#1796] [Everyday usage]
[This blog](https://eric.blog/2019/01/12/install-unison-2-48-4-on-mac-os-x-with-homebrew/) used to be our guideposts for installing this version of unison, but recently it looks like some packages have been marked as outdated and can no longer be installed via the methods outlined. Many google searches and various hacks and attempts haven't been fruitful either. Any advice on how to install older versions of unison for macOS?

**Comment by SMillerDev** [comment]
You can use https://docs.brew.sh/Manpage#extract---version---force-formula-tap to copy a version to a personal tap.

https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap can help you with taps.

**Reply by monk-wellington** [reply]
Thanks for the response. Do I need specific access to older repos?
![Screen Shot 2021-07-07 at 2 57 41 PM](https://user-images.githubusercontent.com/54697724/124833929-bfda1a00-df33-11eb-9994-bf54b5d42a54.png)

**Reply by carlocab** [reply]
No. You need to first create your own tap with something like
```sh
brew tap-new monk-wellington/your-tap-name-here
```
This will create a tap (Git repository) at `$(brew --repository monk-wellington/your-tap-name-here)`. You can then extract the version of Unison that you need with
```sh
brew extract --version=2.48.4 unison monk-wellington/your-tap-name-here
```
You'll want to see `man brew`, `brew help extract`, and `brew help tap-new` for details.

**Reply by monk-wellington** [reply]
Ok. Do I need to file an issue with homebrew/core tap then?

![Screen Shot 2021-07-07 at 3 13 44 PM](https://user-images.githubusercontent.com/54697724/124835355-0a5c9600-df36-11eb-8850-e821971a0679.png)

**Reply by carlocab** [reply]
This is a Homebrew/brew bug. You can file an issue there (not in Homebrew/homebrew-core).

**Reply by nandahkrishna** [reply]
The version you're trying to extract is quite old, and some of the APIs used have undergone significant changes. You could try this instead:
```
brew tap gromgit/dev   # tap @gromgit's dev repo
brew yank --help       # usage instructions

brew yank unison 2.48.4 monk-wellington/unison
brew install monk-wellington/unison/unison@2.48.4
```
The command and Tap are third-party tools and not maintained by Homebrew, but they would be useful in your case.

**Reply by monk-wellington** [reply]
Almost!! Good advice though. 😅

![Screen Shot 2021-07-07 at 3 24 35 PM](https://user-images.githubusercontent.com/54697724/124836321-a5a23b00-df37-11eb-971a-f7e4ddd81ab0.png)

**Reply by carlocab** [reply]
You'll need to `brew edit unison@2.48.4` to update the syntax so that it works with the current version of `brew`.

By the way, please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing the output of commands in your terminal. They're often difficult to read and impossible to reuse.

**Reply by nandahkrishna** [reply]
`unison` 2.48.4 is from 2016, and Homebrew has undergone _many_ changes since then. You will have to edit the Formula to fix any errors you encounter, before installing it.

As Carlo mentioned, you may wish to open an issue at Homebrew/brew for the problem with `brew extract`.

**Reply by monk-wellington** [reply]
Understood. Thanks for your work and your help 👍🏽
